src/java.base/share/classes/jdk/internal/util/Preconditions.java

```diff
@@ -217,39 +217,29 @@ public X apply(String checkKind, List<Number> args) {
 
     private static String outOfBoundsMessage(String checkKind, List<? extends Number> args) {
         if (checkKind == null && args == null) {
-            return String.format("Range check failed");
+            return "Range check failed";
         } else if (checkKind == null) {
-            return String.format("Range check failed: %s", args);
+            return "Range check failed: " + args;
         } else if (args == null) {
-            return String.format("Range check failed: %s", checkKind);
+            return "Range check failed: " + checkKind;
         }
 
-        int argSize = 0;
+        int argSize;
         switch (checkKind) {
-            case "checkIndex":
-                argSize = 2;
-                break;
-            case "checkFromToIndex":
-            case "checkFromIndexSize":
-                argSize = 3;
-                break;
-            default:
+            case "checkIndex" -> argSize = 2;
+            case "checkFromToIndex", "checkFromIndexSize" -> argSize = 3;
+            default -> argSize = 0;
         }
```

Comment:
This can use a switch expression:
```suggestion
        int argSize = switch (checkKind) {
            case "checkIndex" -> 2;
            case "checkFromToIndex", "checkFromIndexSize" -> 3;
            default -> 0;
        }
```